server/modules/record/service/update/thread/recordUpdateThread.js

```diff
@@ -53,12 +58,19 @@ class RecordUpdateThread extends Thread {
       this.postMessage({ type, content })
   }
 
-  async handleNodesUpdated (updatedNodes) {
+  async handleNodesUpdated (updatedNodes, t) {
     this._postMessage(WebSocketEvents.nodesUpdate, updatedNodes)
   }
 
-  async handleNodesValidationUpdated (validations) {
-    this._postMessage(WebSocketEvents.nodeValidationsUpdate, validations)
+  async handleNodesValidationUpdated (validations, t) {
+    const record = await this.getRecord(t)
```

Comment:
i don't think it's a good idea having the transaction in the handleCallbacks.. don't you have already the record in the thread?